### PR TITLE
Add option to not measure ResizingTextureView based on aspect ratio

### PR DESCRIPTION
**Pull request by eygraber**
###### Fixes issue #164.
- [x] This pull request follows the coding standards

###### This PR changes:
 -  Adds an option to not use aspect ratio when measuring the `VideoViewApi`

### CODE REVIEW

File: library/src/main/java/com/devbrackets/android/exomedia/core/video/scale/MatrixManager.java [hunk 1]
```diff
@@ -97,7 +97,9 @@ protected void applyCenterCrop(@NonNull TextureView view, @NonNull Matrix transf
         float yScale = (float)view.getHeight() / intrinsicVideoSize.y;
 
         float scale = Math.max(xScale, yScale);
-        applyScale(view, transformMatrix, scale, scale);
+        xScale = scale / xScale;
```
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/core/video/scale/MatrixManager.java]: Why are we dividing the calculate maxScale?  This will cause an inconsistent scaling (stretching)

File: library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java [hunk 2]
```diff
@@ -94,6 +94,8 @@
 
     void setScaleType(@NonNull ScaleType scaleType);
 
+    void setIgnoreAspectRatio(boolean ignoreAspectRatio);
```
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: I think this is a little vague... for what reason would I "IgnoreAspectRation()" (coming from an external users point of view).  Maybe `setIgnoreAspectRatioInLayout()`?

File: library/src/main/java/com/devbrackets/android/exomedia/core/video/scale/MatrixManager.java (hunk 1 above)
Comment (eygraber) [inline library/src/main/java/com/devbrackets/android/exomedia/core/video/scale/MatrixManager.java]: On the contrary, we need the max of the width and height ratios to use as a normalizing scale to apply to those ratios before they can be used to scale. If we didn't normalize them they could potentially stretch.

Consider this case:

```java
videoWidth = 720;
videoHeight = 1280;
viewWidth = 1440;
viewHeight = 1446; // I know this is an odd number :-)
```

Using the current code, the scale would be `2.0` for the x and y, which does not match the ratio of the video size, so it gets stretched. Using the proposed change results in a scale of `1` for the x, and `1.7704011065` for the y, which matches the ratio between the video size (slightly correcting for the differences in view size).

I just noticed the same issue in the other scaling calculations. I'm not sure if I'm explaining it correctly, [but this project has done a great job of implementing scale](https://github.com/yqritc/Android-ScalableVideoView/blob/master/library/src/main/java/com/yqritc/scalablevideoview/ScaleManager.java). Perhaps this is better done as part of a separate pull request that fixes all the scale calculations.

File: library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java (hunk 2 above)
Comment (eygraber) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: Fair point. I'll see what I can come up with.

File: library/src/main/java/com/devbrackets/android/exomedia/core/video/scale/MatrixManager.java (hunk 1 above)
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/core/video/scale/MatrixManager.java]: It should be a separate PR.  To me it still seems wrong, but it has been a while since I've done manual matrix multiplication so maybe I'm remembering something incorrectly.

File: library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java (hunk 2 above)
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: Thanks

File: library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java [hunk 3]
```diff
@@ -94,6 +94,8 @@
 
     void setScaleType(@NonNull ScaleType scaleType);
 
+    void setDoNotMeasureBasedOnAspectRatio(boolean doNotMeasureBasedOnAspectRatio);
```
Comment (eygraber) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: @brianwernick How about this?

Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: Hmm, typically methods that take booleans (toggle methods) are worded towards enabling something (e.g. `setStateEnabled(...)`); would you re-word it to follow that paradigm?  

e.g.
```java
void setAspectAwareOnMeasureEnabled(boolean enabled);
// or 
void setMeasureBasedOnAspectRatioEnabled(boolean enabled);
// or something similar
```

I know I'm being a little picky on this so thanks for hanging in there :)

Comment (eygraber) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: No problem

Comment (eygraber) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: Done
